daemon/exec_linux.go

```diff
@@ -23,5 +23,27 @@ func execSetPlatformOpt(c *container.Container, ec *exec.Config, p *libcontainer
 	if ec.Privileged {
 		p.Capabilities = caps.GetAllCapabilities()
 	}
+	if apparmor.IsEnabled() {
+		var appArmorProfile string
+		if c.AppArmorProfile != "" {
+			appArmorProfile = c.AppArmorProfile
+		} else if c.HostConfig.Privileged {
```

Comment:
Indeed, the privileged model for exec is non obvious. `hostconfig` is in fact correct - `docker exec --privileged` does not in fact disable apparmor at present.